src/storage/txn/scheduler.rs

```diff
@@ -546,6 +563,59 @@ impl Scheduler {
         extract_ctx(ctx.cmd.as_ref().unwrap())
     }
 
+    fn statistic(&mut self, cmd: &Command) {
+        match *cmd {
+            Command::Prewrite { ref ctx, ref mutations, .. } => {
+                let region_id = ctx.get_region_id();
+                self.write_stats.incr(region_id, mutations.len() as u64);
+            }
+            _ => debug!("don't need statistic for command {:?}", cmd),
+        }
+    }
+
+    fn should_schedule_gc(&mut self, ctx: &Context) -> bool {
+        if !self.pro_exec_gc {
+            return true;
+        }
+
+        let region_id = ctx.get_region_id();
+        let mut history_write = self.write_stats.get_history(region_id);
+        history_write >>= 4;
+        let pro: u8 = if history_write >= 255 {
+            255
+        } else {
+            history_write as u8
+        };
+
+        let x = rand::random::<u8>();
+        let res = pro > x;
+        if res {
+            self.write_stats.clear_history(region_id);
+        }
```

Comment:
We check if a command should schedule just when receive a new command from the outside of scheduler (`on_receive_new_cmd`). If a GC command split into many new commands, we directly schedule it(`schedule_cmd`).